exonum-light-client/src/main/java/com/exonum/client/ExonumHttpClient.java

```diff
@@ -106,6 +115,68 @@ public String getUserAgentInfo() {
     });
   }
 
+  @Override
+  public BlockResponse getBlockByHeight(long height) {
+    HttpUrl url = urlBuilder()
+        .encodedPath(BLOCK)
+        .addQueryParameter("height", String.valueOf(height))
+        .build();
+    Request request = getRequest(url);
+
+    return blockingExecuteAndParse(request, ExplorerApiHelper::parseGetBlockResponse);
+  }
+
+  @Override
+  public BlocksResponse getBlocks(int count, boolean skipEmpty, Long heightMax,
+      boolean withBlocksTime) {
+    return doGetBlocks(count, skipEmpty, heightMax, withBlocksTime);
+  }
+
+  @Override
+  public BlocksResponse getLastBlocks(int count, boolean skipEmpty, boolean withBlocksTime) {
+    return doGetBlocks(count, skipEmpty, null, withBlocksTime);
+  }
+
+  @Override
+  public Block getLastBlock() {
+    BlocksResponse response = doGetBlocks(1, false, null, false);
+
+    return response.getBlocks()
+        .stream()
+        .findFirst()
+        .orElseThrow(() -> new AssertionError("Should never happen"));
+  }
+
+  @Override
+  public Optional<Block> getLastNotEmptyBlock() {
```

Comment:
nonEmpty? (@Maria-Nosyk is that correct?)

Reply:
`NonEmpty` is correct